lib/ouija_tests.ml

```diff
@@ -1,74 +1,96 @@
-open OUnit2
+open Core.Std
+open Async.Std
+open Sentinel.Std
 
-let rest _ =
+let check exp op = fun _ -> return (exp = op ())
```

Comment:
Ok, that is a problem of lacking documentation :)

The point of Basic vs Async is not really whether the fold is sequential or parallel (they are both sequential currently anyway) but whether the tests lives within the Async monad.

The main problem I want to solve with sentinel is that you can't really tests things that return Async.t with "traditional" tools like Ounit. Given that, instead of writing a tool specific for Async I approached it trying to make a tool that can work with arbitrary monads (with the assumption that Monads are expressive enough to represent any possible outcome).

But given that, if you don't need Async there is no need to bring it in just for testing (even though you are going to needed it for the runner, but that is a different thing).

The way sentinel folds the suites now is very likely to change in the future towards a parallel approach, so you should not assume any kind of dependency between the different tests. Actually I consciously avoided threading the state through the tests to make impossible (or at least more difficult :) ) to affect one test from the others.